plutus-core/plutus-core/examples/PlutusCore/Examples/Builtins.hs

```diff
@@ -177,34 +181,40 @@ data BuiltinErrorCall = BuiltinErrorCall
 --    to be handled correctly by design
 instance uni ~ DefaultUni => ToBuiltinMeaning uni ExtensionFun where
     type CostingPart uni ExtensionFun = ()
-    toBuiltinMeaning :: forall val. HasMeaningIn uni val => ExtensionFun -> BuiltinMeaning val ()
 
-    toBuiltinMeaning Factorial =
+    data BuiltinVersion ExtensionFun = ExtensionFunV0 | ExtensionFunV1
```

Comment:
Why not start at 1 like the other version numbers?

Also, the purpose of this isn't clear to me.  Is it in case we later do want version-dependent things here?

Reply:
I started with 0 to show that the numbers are meaningless and you can use whatever you want.
This is an example fun and if you see later on, I use it in some "dumb" builtin to reflect this number. It is just a showcase and for testing purpose.

But you are right in the fact that this approach enforces every possible `fun` to have versioning, although it is not needed (e.g. see Nops, another example fun)